src/builder.rs

```diff
@@ -331,7 +326,9 @@ impl ClientBuilder {
             let (sync_kill_sender, sync_kill_receiver) = channel(1);
             let runtime = std::thread::spawn(move || {
                 let runtime = Runtime::new().expect("failed to create Tokio runtime");
-                runtime.block_on(Client::sync_nodes(&sync_, &nodes, &network_info_));
+                runtime
+                    .block_on(Client::sync_nodes(&sync_, &nodes, &network_info_))
+                    .expect("TODO");
```

Comment:
If you want to keep it as todo in this PR, at least add something like `block_on failed`

Reply:
What do you think is acceptable here ? We're on a thread. I could get the error and panic if `synced` is false ?

Reply:
Why only if ` if synced is false`? Even if it's synced at that moment, it could get unsynced at a later point

Reply:
So just a warn log like we do in the spawned task?

Reply:
Below we have `.expect("failed to init node syncing process")`, not sure if just a warning is good if you then maybe work with unsynced nodes